code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -688,94 +688,56 @@
-				return
-			var/working_state = ""
-			switch(input)
-				if("poly")
-					working_state = "parrot_fly"
-				if("chicken")
-					working_state = "chicken_brown"
-				if("spider")
-					working_state = "guard"
-				else
-					working_state = input
-			hologram_appearance = mutable_appearance(icon_list[input], working_state)
-		else
-			var/list/icon_list = list(
-				"default" = 'icons/mob/silicon/ai.dmi',
-				"floating face" = 'icons/mob/silicon/ai.dmi',
-				"xeno queen" = 'icons/mob/nonhuman-player/alien.dmi',
-				"horror" = 'icons/mob/silicon/ai.dmi',
-				"clock" = 'icons/mob/silicon/ai.dmi'
-				)
-
-			input = tgui_input_list(usr, "Select a hologram", "Hologram", sort_list(icon_list))
-			if(isnull(input))
-				return
-			if(isnull(icon_list[input]))
-				return
-			var/working_state = ""
-			switch(input)
-				if("xeno queen")
-					working_state = "alienq"
-				else
-					working_state = input
-			hologram_appearance = mutable_appearance(icon_list[input], working_state)
+
+	var/static/list/choices = assoc_to_keys(GLOB.ai_hologram_category_options) + "Existing Character"
+	var/choice = tgui_input_list(usr, "What kind of hologram do you want?",	"Customize", choices)
+
+	if(choice == "Existing Character")
+		switch(tgui_alert(usr,"Would you like to base it off of your current character loadout, or a member on station?", "Customize", list("My Character","Station Member")))
+			if("Station Member")
+				var/list/personnel_list = list()
+
+				for(var/datum/record/locked/record in GLOB.manifest.locked)//Look in data core locked.
+					personnel_list["[record.name]: [record.rank]"] = record.character_appearance//Pull names, rank, and image.
+				if(!length(personnel_list))
+					tgui_alert(usr,"No suitable records found. Aborting.")
+					return
+
+				var/input = tgui_input_list(usr, "Select a crew member", "Station Member", sort_list(personnel_list))
+				if(isnull(input))
+					return
+				if(isnull(personnel_list[input]))
+					return
+				var/mutable_appearance/character_icon = personnel_list[input]
+				if(character_icon)
+					character_icon.setDir(SOUTH)
+					hologram_appearance = character_icon
+
+			if("My Character")
+				switch(tgui_alert(usr,"WARNING: Your AI hologram will take the appearance of your currently selected character ([usr.client.prefs?.read_preference(/datum/preference/name/real_name)]). Are you sure you want to proceed?", "Customize", list("Yes","No")))
+					if("Yes")
+						var/mob/living/carbon/human/dummy/ai_dummy = new
+						var/mutable_appearance/dummy_appearance = usr.client.prefs.render_new_preview_appearance(ai_dummy)
+						if(dummy_appearance)
+							qdel(ai_dummy)
+							hologram_appearance = dummy_appearance
+					if("No")
+						return FALSE
+	else
+		var/list/options = GLOB.ai_hologram_category_options[choice]
+		var/option = tgui_input_list(usr, "Select a hologram", "Hologram", options)
+		if(!option)
+			return
+
+		var/icon = GLOB.ai_hologram_icons[option]
+		if(!icon)
+			return
```

Comment:
Maybe crash here so that there's some kind of record that someone failed to update one of the lists.